SpatialGDK/Documentation/content/non-unreal-layers.md

```diff
@@ -129,48 +128,31 @@ Worker_RequestId SendSomeCommandRequest(Worker_EntityId TargetEntityId, Worker_C
 You could receive and deserialize a component update and command request in your Unreal project code in the following way:
 ```
-UCLASS()
-class SPATIALGDK_API SessionComponentCallbacks : UOpCallbackTemplate
+void ATPSGameMode::BeginPlay()
 {
-    GENERATED_BODY()
-
-public:
-    Worker_ComponentId GetComponentId()
-    {
-        return 1337;
-    }
-
-    void OnComponentUpdate(const Worker_ComponentUpdateOp& Op) override
-    {
-        Worker_ComponentUpdate Update = {};
-        Update.component_id = 1338;
-        Update.schema_type = Schema_CreateComponentUpdate(1338);
-        Schema_Object* FieldsObject = Schema_GetComponentUpdateFields(Update.schema_type);
-        Schema_AddInt32(FieldsObject, 1, ++UnrealCounter);
-        Cast<USpatialNetDriver>(World->GetNetDriver())->Connection-> SendComponentUpdate(39, &Update);
-    }
-
-    void OnCommandRequest(const Worker_CommandRequestOp& Op) override
-    {
+    UWorld* World = GetWorld();
+    USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(World->GetNetDriver());
+    USpatialDispatcher* Dispatcher = NetDriver->Dispatcher;
+
+    Dispatcher->AddOpCallback(1337, [&](const Worker_ComponentUpdateOp& Op) {
+        uint32 PlayerCount = Schema_GetUint32(Schema_GetComponentUpdateFields(Op.update.schema_type), 1);
+        UE_LOG(LogTPS, Verbose, TEXT("Received update, new player count: %d"), PlayerCount);
+        const FVector = FVector::ZeroVector;
+        const FRotator Rotation = FRotator::ZeroRotator;
+        GetWorld()->SpawnActor(CubeClass, &Location, &Rotation);
+    });
+
+    Dispatcher->AddOpCallback(1338, [&](const Worker_CommandRequestOp& Op) {
         Worker_CommandResponse Response = {};
         Response.component_id = 1338;
         Response.schema_type = Schema_CreateCommandResponse(1338, 1);
-        Schema_Object* ResponseObject = Schema_GetCommandResponseObject(Response.schema_type);
+        Schema_Object* response_object = Schema_GetCommandResponseObject(Response.schema_type);
 
-        const char* Text = "My Response text.";
-        Schema_AddBytes(ResponseObject, 1, (const uint8_t*)Text, sizeof(char) * strlen(Text));
+        FString text = "Here's my response.";
+        Schema_AddBytes(response_object, 1, (const uint8_t*) TCHAR_TO_ANSI(*text), sizeof(char) * strlen(TCHAR_TO_ANSI(*text)));
 
-        // Spawn an actor using the World pointer
-        // (where CubeClass is a Blueprint Actor UClass pointer)
-        if (StaticComponentView->HasAuthority(Op.entity_id, GetComponentId()))
-        {
-            const FVector Location(0.0f, 0.0f, 0.0f);
-            const FRotator Rotation = FRotator::ZeroRotator;
-            World->SpawnActor(CubeClass, &Location, &Rotation);
-        }
-
-        Cast<USpatialNetDriver>(World->GetNetDriver())->Connection->SendCommandResponse(Op.request_id, &Response);
-    }
+        Cast<USpatialNetDriver>(GetWorld()->GetNetDriver())->Connection->SendCommandResponse(Op.request_id, &Response);
+    });
 }
 ```
```

Comment:
Don't forget to update the metadata with the date and the update.

`2019-04-04 Page updated with full editorial review: updated methods for receiving network operations`